modules/integration/tests-integration/tests-mqtt/src/test/java/org/wso2/mb/integration/tests/mqtt/functional/PermissionTestCase.java

```diff
@@ -0,0 +1,205 @@
+/*
+*  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.wso2.mb.integration.tests.mqtt.functional;
+
+
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.FrameworkConstants;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.context.beans.User;
+import org.wso2.mb.integration.common.clients.ClientMode;
+import org.wso2.mb.integration.common.clients.MQTTClientConnectionConfiguration;
+import org.wso2.mb.integration.common.clients.MQTTClientEngine;
+import org.wso2.mb.integration.common.clients.MQTTConstants;
+import org.wso2.mb.integration.common.clients.QualityOfService;
+import org.wso2.mb.integration.common.utils.backend.MBIntegrationBaseTest;
+
+import javax.xml.xpath.XPathExpressionException;
+import java.util.List;
+
+/**
+ * The following test cases are related to permissions of users and tenant domains.
+ */
+public class PermissionTestCase extends MBIntegrationBaseTest {
+
+    /**
+     * Initialize super class.
+     *
+     * @throws XPathExpressionException
+     */
+    @BeforeClass(alwaysRun = true)
```

Comment:
Have you considered DataProviders if using data providers doesn't add value lets not add that... just  a suggestion.

[1] http://testng.org/doc/documentation-main.html